src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritHandler.java

```diff
@@ -84,33 +94,104 @@
      * Creates a GerritHandler with all the default values set.
      *
      * @see GerritDefaultValues#DEFAULT_NR_OF_RECEIVING_WORKER_THREADS
+     * @see GerritDefaultValues#DEFAULT_RECEIVE_THREAD_KEEP_ALIVE_TIME
      */
     public GerritHandler() {
-        this(DEFAULT_NR_OF_RECEIVING_WORKER_THREADS);
+        this(DEFAULT_NR_OF_RECEIVING_WORKER_THREADS, DEFAULT_RECEIVE_THREAD_KEEP_ALIVE_TIME);
     }
 
     /**
-     * Standard Constructor.
+     * Create handler with the given number of maximum worker threads
      *
      * @param numberOfWorkerThreads the number of event threads.
      */
     public GerritHandler(int numberOfWorkerThreads) {
+        this(numberOfWorkerThreads, DEFAULT_RECEIVE_THREAD_KEEP_ALIVE_TIME);
+    }
+
+    /**
+     * Create handler with the given number of maximum worker threads and given thread keep alive
+     * time.
+     *
+     * @param numberOfWorkerThreads the number of event threads.
+     * @param threadKeepAliveTime the number of seconds threads will stay alive.
+     */
+    public GerritHandler(int numberOfWorkerThreads, int threadKeepAliveTime) {
         this.numberOfWorkerThreads = numberOfWorkerThreads;
+        this.threadKeepAliveTime = threadKeepAliveTime;
```

Comment:
Since `executor.allowCoreThreadTimeOut(true);` is specified, I think we need to specify some minimum safe number here. According to the documentation setting for example the timeout to 0 will cause new threads to be created all the time and thrown away upon completion. And I think that might not be desirable.

Reply:
what would be a reasonable minimum value? 30 seconds? Maybe we could also use a fixed value (not allowing it to be changed but that makes testing harder and when you reduce the number of threads it might take quite long until they die.

Reply:
I think we can go lower, even 10 seconds. IIUC as long as it's not `0` the core thread count should stay the same.